crates/nu-command/tests/commands/first.rs

```diff
@@ -98,3 +111,15 @@ fn errors_on_negative_rows() {
 
     assert!(actual.err.contains("use a positive value"));
 }
+
+#[test]
+fn errors_on_empty_list_when_no_rows_given() {
+    let actual = nu!(pipeline(
+        r#"
+            []
+            | first
+        "#
+    ));
+
+    assert!(actual.err.contains("index too large"));
+}
```

Comment:
Thanks for diligently including tests for your changes @baehyunsol 

Putting my "Semantics, Semantics, Semantics" head on for a moment I come with a question not intended as criticisism of your changes but as a question for all of us reviewing this:

WHY should pulling `| first` from an empty list or liststream error out. We should be able to explain to our users what the rules are when commands fail if presented with empty or no data. From a pure typechecking perspective this seems very valid on the input side but could become muddied on the output side.

As context for `first` we have the similar `take` which has a simpler behavior assuming to always operate on iterable types and returning iterables. (so no magic `first` without argument that unpacks) But if we say that we are fine with an error on `first` would we be fine with an error on `last` where there is no direct equivalent like `take`

Reply:
I thought `first` should behave like `get 0` and `first n` should behave like `take n`. Since, `[] | get 0` is an error and `[] | take 1` is an empty list, I think that's consistent.

I also think `first` and `last` should behave in similar way, so I implemented that way. What's breaking the consistency is that `first` has its partner `take`, but `last` doesn't. I think the best way would be to implement a counterpart of `last`. Using negative numbers with `take`, as mentioned in #6611, could be a solution.